### ISSUE TITLE
This needs to be more dynamic

### ISSUE THREAD

**Issue by danneu** [danneu/cloudflare-ip#6]
If Cloudflare adds ip addresses to their public list and you don't add them to your whitelist, then users will get denied. This library should offer something more dynamic.

**Comment by ErisDS** [comment]
I would suggest something like:

- expose the update script as something that can be easily called from cron / other tools
- make the location of the list file configurable

That way I can control the file outside of git, but it would still be backwards compatible.

**Comment by danneu** [comment]
Agreed. It also needs to expose a way to both fetch new addresses and change the list in-process. 

That way a server can just run a concurrent loop at some interval that updates the list with no downtime nor out-of-process dependency.

I think this could be established with an API like this:

```javascript
const CloudflareIp = require('cloudflare-ip')

CloudflareIp.fetchList() //-> Promise<Array>

// If no path given, falls back to library's internal list
const cip = new CloudflareIp('/home/user/cloudflare-ips.txt')

cip.isCloudflareIp('1.2.3.4') //-> Boolean
cip.setList([...])
```

Concurrent update loop, maybe provided by `cip.updateLoop(delay)` since it's the 99% case:

```javascript
async function updateLoop(cip, delay) {
  const timeout = (delay) => new Promise((resolve) => setTimeout(resolve, delay))

  while (true) {
    let list
    try {
      list = await CloudflareIp.fetchIps()
    } catch (e) {
      warn('could not fetch ip list from cloudflare', e)
    }

    if (list) {
      info('updated cloudflare ip list')
      cip.setList(list)
    }

    await timeout(delay)
  }
}
```

Out of process update script:

    echo 'https://www.cloudflare.com/ips-v4\nhttps://www.cloudflare.com/ips-v6' | xargs curl > /home/user/cloudflare-ips.txt
    
Or, via library's cross-plat helper:

    ./node_modules/.bin/fetch-cloudflare-ips > cloudflare-ips.txt
